src/app/rpmostree-compose-builtin-tree.c

```diff
@@ -272,13 +272,13 @@ install_packages_in_root (RpmOstreeTreeComposeContext  *self,
   if (opt_cachedir && !opt_unified_core)
     dnf_context_set_keep_cache (dnfctx, TRUE);
   /* For compose, always try to refresh metadata; we're used in build servers
-   * where fetching should be cheap. Otherwise, if --cache-only is set, it's
-   * likely an offline developer laptop case, so never refresh.
+   * where fetching should be cheap. Otherwise, if --cache-only is set, well...
+   * today that just basically sets the default of honoring the repo
+   * metadata_expire. We'd need to enhance the libdnf to have a "really never
+   * expire" option.
    */
   if (!opt_cache_only)
-    dnf_context_set_cache_age (dnfctx, 0);
-  else
-    dnf_context_set_cache_age (dnfctx, G_MAXUINT);
+    rpmostree_context_set_expire_pkgcache (self->corectx);
```

Comment:
Don't we still want `G_MAXUINT` in the treecompose path for the developer laptop case, though? (And actually... for `coreos-assembler build` too, right?).

So basically:

- treecompose path without `--cache-only` --> 0
- treecompose path with `--cache-only` --> `G_MAXUINT`
- layering path --> `G_MAXUINT-1`

?

Reply:
Ooh yep, thanks for catching all of this!  Updated now, only lightly tested :arrow_down: